### ISSUE TITLE
Global variable that depends on another global variable from another file fails intermittently.

### ISSUE THREAD

**Issue by hemikak** [ballerina-platform/ballerina-lang#8546]
**Description:**
When a global variable that is initialized using a function uses a global variable from another bal file, the behavior differs when running the application multiple times.

**Steps to reproduce:**
Assume to following 2 files in a package

constants.bal
----------------
```ballerina
import ballerina/config;

@final int SERVICE_PORT = config:getAsInt("SERVICE_PORT", default = 9090);
```

my_function.bal
----------------
```ballerina
@final int INCREMENTED_SERVICE_PORT = incrementPort();

function incrementPort() returns (int) {
    return SERVICE_PORT + 1;
}
```

Running the above package behaves in 2 different ways.
1. The constant.bal file is first loaded. Then the my_function.bal file loaded. Having so the INCREMENTED_SERVICE_PORT will have the incremented value of SERVICE_PORT. This is the happy path.

2. my_function.bal file is loaded first and then constants.bal file is loaded. INCREMENTED_SERVICE_PORT will have no value as SERVICE_PORT is not yet defined.

The behavior is intermittently different. Also note that I want to use SERVICE_PORT variable in multiple other files and function etc. Hence the requirement is that it should be a global. I do not want to redefine it as another global variable in other files.

**Affected Versions:**
0.970.0

**OS, DB, other environment details and versions:**
OSX, Ubuntu

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by MaryamZi** [comment]
@rdhananjaya, shouldn't this be resolved with https://github.com/ballerina-platform/ballerina-lang/pull/13094?

**Comment by rdhananjaya** [comment]
Yes this is fixed.

**Comment by Shan1024** [comment]
Closing the issue since this is already fixed with #13094.
